examples/oauth_producer.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# This uses OAuth client credentials grant:
+# https://www.oauth.com/oauth2-servers/access-tokens/client-credentials/
+# where client_id and client_secret are passed as HTTP Authorization header
+#
+
+import logging
+import functools
+import argparse
+import time
+from confluent_kafka import SerializingProducer
+from confluent_kafka.serialization import StringSerializer
+import requests
+
+
+def _get_token(args, config):
+    """Note here value of config comes from sasl.oauthbearer.config below.
+    It is not used in this example but you can put arbitrary values to
+    configure how you can get the token (e.g. which token URL to use)
+    """
+    payload = {
+        'grant_type': 'client_credentials',
+        'scope': ' '.join(args.scopes)
+    }
+    resp = requests.post(args.token_url,
+                         auth=(args.client_id, args.client_secret),
+                         data=payload)
+    token = resp.json()
+    return token['access_token'], time.time() + float(token['expires_in'])
+
+
+def producer_config(args):
+    logger = logging.getLogger(__name__)
+    return {
+        'bootstrap.servers': args.bootstrap_servers,
+        'key.serializer': StringSerializer('utf_8'),
+        'value.serializer': StringSerializer('utf_8'),
+        'security.protocol': 'sasl_plaintext',
+        'sasl.mechanisms': 'OAUTHBEARER',
+        'sasl.oauthbearer.config': 'not-important',
```

Comment:
nit: this make it seem like sasl.oauthbearer.config is not generally important, but required.
Perhaps better to make it more explicit by commenting out the line with a comment saying that no extra configuration is required for this example, but that the property is used to pass configuration to the oauth_cb.